### DISCUSSION TITLE
Airflow Context in KubernetesPodOperator Callback

### DISCUSSION THREAD

**Discussion by gffryclrk** [apache/airflow#41202] [Q&A]
Hi Airflow Community,

I'm new to the community and I have a question regarding [KubernetesPodOperator Callbacks](https://airflow.apache.org/docs/apache-airflow-providers-cncf-kubernetes/stable/operators.html#kubernetespodoperator-callbacks): is it possible to access the Airflow context from within these callback functions? I'd like to access task id, dag id & a few other context variables in the callback function but it seems that context [isn't passed as an argument](https://github.com/apache/airflow/blob/9fbaca0ab0150d899caea172ea3d1085982edd4f/airflow/providers/cncf/kubernetes/operators/pod.py#L613-L615) when the callbacks are called. This seems like a straightforward enough thing to add and I'm willing to create a PR but wanted to check and see if I was missing something first before doing so. Please let me know!

Thank you,

Geoffrey

**Comment by olivd4400** [comment]
Hello,

Did you get any news? I would also like to access context task instance to be able to push into xcom from those callbacks.

Regards,
Olivier

**Reply by gffryclrk** [reply]
Hi @olivd4400 , thank you for following up. I haven't heard any more about this but I think it would be useful. I'm going to look more into it in the coming weeks and will follow up on this thread when I have more information. Please let me know if you learn more. Thank you.

**Comment by perry2of5** [comment]
Looks like this is implemented in the latest versions:

https://github.com/apache/airflow/blob/main/providers/cncf/kubernetes/src/airflow/providers/cncf/kubernetes/operators/pod.py#L675

https://github.com/apache/airflow/blob/main/providers/cncf/kubernetes/src/airflow/providers/cncf/kubernetes/callbacks.py#L63

**Reply by gffryclrk** [reply]
Thank you @perry2of5 for pointing this out! It's very useful.

**Comment by gffryclrk** [comment]
Implemented. See [above](https://github.com/apache/airflow/discussions/41202#discussioncomment-14798500).
